### PR TITLE
build improvements

### PR DESCRIPTION
**Pull request by philvarner**
Numerous build improvements:

- add 3.9 and 3.10 to the text matrix
- update black, isort, and flake8 versions
- add instructions to install pre-commit hooks and how to run tox
- remove all unused imports
- change line length to 120 b/c there are a lot that are longer than 90
- fixup warnings from flake8
- not sure why numpy was a dep for running tox?
- add config for running mypy, but leave commented since there are a number of problems to fix

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+    flake8 .
+    black .
+;    mypy stac_pydantic/
 
 # Lint
 [flake8]
-exclude = .git,__pycache__,docs/source/conf.py,old,build,dist
-max-line-length = 90
+exclude = .git,__pycache__,docs/source/conf.py,old,build,dist,.venv,.tox,.idea
+max-line-length = 120
```
Comment (kylebarron) [inline tox.ini:20]: Regardless of the value here, `black` defaults to `88`, so they suggest setting the flake8 config to 88: https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length

Comment (philvarner) [reply tox.ini]: I was wondering about that -- black wasn't giving an error but flake8 was on those lines. I've updated the flake8 config per the black documentation.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
-      - id: isort
+      - id: flake8
         language_version: python3
+
+#  - repo: https://github.com/pre-commit/mirrors-mypy
+#    rev: v0.931
+#    hooks:
+#      - id: mypy
+#        exclude: ^tests/
+#        args: [--strict]
```
Comment (moradology) [inline .pre-commit-config.yaml:26]: Is this meant to be commented out?

**Comment by moradology** [comment]
I believe there will still be one failing test. This should be resolved once the next version of geojson-pydantic is released!

**Review by geospatial-jeff — COMMENTED** [review]
The build changes look good to me, although there are other breaking changes sprinkled through the PR (like removing of `Link.__iter__`) and several others which means we will have to release as a new major version.

I think I would like to separate the build improvements from the functionality change to isolate the breaking code changes in a different PR.

**Comment by moradology** [comment]
After further discussion, it looks like the most sensible move might be to accept this PR after cutting a minor release. This will avoid the complexities of getting this PR's non-breaking bits in without the breaking portions.

**Review by geospatial-jeff — APPROVED** [review]

**Pull request merged**